packages/@react-aria/overlays/src/useModal.tsx

```diff
@@ -118,13 +118,20 @@ export function OverlayContainer(props: ModalProviderProps): React.ReactPortal {
   return ReactDOM.createPortal(contents, document.body);
 }
 
+interface ModalAria {
+  modalProps: {
+    /** Data attribute marks the dom node as a modal for the aria-modal-polyfill. */
+    'data-ismodal': boolean
+  }
+}
```

Comment:
I think this should be opaque like the rest of our hooks.

```suggestion
  /** Props for the modal content element. */
  modalProps: HTMLAttributes<HTMLElement>
}
```